CSharpProvider/tests/compare.py

```
@@ -0,0 +1,316 @@
+
+        # ── Rust provider ──
+        print(f"\n--- Rust provider (port {RUST_PORT}) ---")
+        start_server(
+            ["cargo", "run", "--", "--port", str(RUST_PORT), "--name", "c-sharp"],
+            RUST_PORT, "Rust provider",
+        )
+
+        print("  Init...")
+        init_resp = run_init_rust(RUST_PORT, str(location))
+        if init_resp and init_resp.get("successful"):
+            print(f"  Init OK")
```

Ruff (0.15.13)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)